### PR TITLE
CODENVY-568: Add possibility handel event then factory fully accepted 

### PR DESCRIPTION
**Pull request by vparfonov**
Start reimport maven model  of Maven projects

### CODE REVIEW

**Comment by vparfonov** [comment]
@skabashnyuk @sleshchenko @mshaposhnik

File: core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/factory/FactoryAcceptedHandler.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.api.factory;
+
+import com.google.gwt.event.shared.EventHandler;
+
+/**
+ * Handel FactoryAcceptedEvent
```
Comment (sleshchenko) [inline core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/factory/FactoryAcceptedHandler.java]: Handel?

File: plugins/plugin-maven/che-plugin-maven-ide/src/main/java/org/eclipse/che/plugin/maven/client/project/MavenModelImporter.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        for (ProjectConfigDto project : projects) {
+            if (MavenAttributes.MAVEN_ID.equals(project.getType()))
+                paths.add(project.getPath());
+        }
+        reimport(paths);
+    }
+
+    /**
+     *
+     * @param projectPaths
```
Comment (sleshchenko) [inline plugins/plugin-maven/che-plugin-maven-ide/src/main/java/org/eclipse/che/plugin/maven/client/project/MavenModelImporter.java:69]: please, fix java docs or remove it at all

**Comment by mshaposhnik** [comment]
Ok

**Comment by skabashnyuk** [comment]
ок

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java [hunk 3]
```diff
@@ -151,7 +151,7 @@ public String downloadSource(@QueryParam("projectpath") String projectPath, @Que
         return Boolean.toString(classpathManager.downloadSources(projectPath, fqn));
     }
 
-    @PUT
+    @GET
```
Comment (benoitf) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java]: any reason to change the operation method ?

Comment (vparfonov) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java]: any reason will be returned back

File: plugins/plugin-maven/che-plugin-maven-ide/src/main/java/org/eclipse/che/plugin/maven/client/service/MavenServerServiceClientImpl.java [hunk 4]
```diff
@@ -83,14 +85,15 @@ public Boolean getPayload() {
...
+    public Promise<Void> reImportProjects(@NotNull List<String> projectsPaths) {
         StringBuilder queryParameters = new StringBuilder();
         for (String path : projectsPaths) {
             queryParameters.append("&projectPath=").append(path);
         }
         final String url = appContext.getDevMachine().getWsAgentBaseUrl() + servicePath + "reimport" +
                            queryParameters.toString().replaceFirst("&", "?");
-        return asyncRequestFactory.createPutRequest(url, null).send();
+
+        return asyncRequestFactory.createPostRequest(url, null).send();
```
Comment (benoitf) [inline plugins/plugin-maven/che-plugin-maven-ide/src/main/java/org/eclipse/che/plugin/maven/client/service/MavenServerServiceClientImpl.java:96]: it should be still createPutRequest no ?

Comment (vparfonov) [inline plugins/plugin-maven/che-plugin-maven-ide/src/main/java/org/eclipse/che/plugin/maven/client/service/MavenServerServiceClientImpl.java:96]: why you dont like POST?

**Pull request merged**